### DISCUSSION TITLE
Headings' ID?

### DISCUSSION THREAD

**Discussion by Lakelimbo** [ssssota/svelte-exmarkdown#90] [Q&A] [answered]
Hello. I really love this library, it's exactly what I need for my website! However I noticed it does not add an ID for each heading element, so it can be used for things like a Table of Contents.
How can I achieve that?

**Comment by ssssota** [comment, marked as answer]
Thanks for your question!

You can use [rehype-slug](https://github.com/rehypejs/rehype-slug) to add `id`s to headings!
And, you can use [remark-toc](https://github.com/remarkjs/remark-toc) to add TOC in your document.

Please check this example!
https://stackblitz.com/edit/svelte-exmarkdown-heading-it-toc?file=src%2Flib%2Fheading-toc-plugin%2Findex.ts

**Reply by Lakelimbo** [reply]
Oh, thanks a lot! I appreciate it.
